src/Symfony/Bundle/FrameworkBundle/Templating/TemplateResponse.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Templating;
+
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @author Pierre du Plessis <pdples@gmail.com>
+ */
+class TemplateResponse implements TemplateResponseInterface
+{
+    private $template;
+    private $parameters;
+    private $statusCode;
+    private $headers;
+
+    public function __construct($template, array $parameters = array(), $statusCode = Response::HTTP_OK, array $headers = array())
```

Comment:
It might make the parameter list too long. What about dropping the `$statusCode` and `$headers` parameters, and include the option to pass a Response object?

```php
public function __construct($template, array $parameters = array(), Response $response = null)
```

That way we can keep it simple to only pass a template and parameters, but if you want more flexibility like setting cookies then you just create your own Response